### PR TITLE
initial implementation

### PR DESCRIPTION
**Pull request by mcalhoun**
## what
* Create a module to provision [AWS ECR Public ](https://docs.aws.amazon.com/AmazonECR/latest/public/public-repositories.html) repos 

## references
* Based on the [terraform-aws-ecr](https://github.com/cloudposse/terraform-aws-ecr-public) module

### CODE REVIEW

File: test/src/examples_complete_test.go [hunk 1]
```diff
@@ -1,78 +1,44 @@
 package test
```
Comment (korenyoni) [inline test/src/examples_complete_test.go:1]: Nice to have: add a disabled test https://github.com/cloudposse/terraform-aws-msk-apache-kafka-cluster/blob/cd11751cb4c999f3c17013dd32336659a277744e/test/src/examples_complete_test.go#L58

File: variables.tf [hunk 2]
```diff
@@ -1,4 +1,54 @@
...
+  architectures = The system architecture that the images in the repository are compatible with. On the Amazon ECR 
+  Public Gallery, the following supported architectures will appear as badges on the repository and are used as search 
+  filters: ARM, ARM 64, x86, x86-64
+
+  operating_systems = The operating systems that the images in the repository are compatible with. On the Amazon ECR 
+  Public Gallery, the following supported operating systems will appear as badges on the repository and are used as 
+  search filters: Linux, Windows
+
+  logo_image_blob = The base64-encoded repository logo payload. (Only visible for verified accounts) Note that drift 
+  detection is disabled for this attribute.
```
Comment (korenyoni) [inline variables.tf]: This formatting is going to look nicer in the markdown, IMO

```suggestion
  `name`: List of Docker local image names, used as repository names for AWS ECR 
  
  `description`: A short description of the contents of the repository. This text appears in both the image details and 
  also when searching for repositories on the Amazon ECR Public Gallery.

  `about_text`: A detailed description of the contents of the repository. It is publicly visible in the Amazon ECR Public 
  Gallery. The text must be in markdown format.

  `usage_text`: Detailed information on how to use the contents of the repository. It is publicly visible in the Amazon 
  ECR Public Gallery. The usage text provides context, support information, and additional usage details for users of 
  the repository. The text must be in markdown format.

  `architectures`: The system architecture that the images in the repository are compatible with. On the Amazon ECR 
  Public Gallery, the following supported architectures will appear as badges on the repository and are used as search 
  filters: ARM, ARM 64, x86, x86-64

  `operating_systems`: The operating systems that the images in the repository are compatible with. On the Amazon ECR 
  Public Gallery, the following supported operating systems will appear as badges on the repository and are used as 
  search filters: Linux, Windows

  `logo_image_blob`: The base64-encoded repository logo payload. (Only visible for verified accounts) Note that drift 
  detection is disabled for this attribute.
```

File: versions.tf [hunk 3]
```diff
@@ -1,11 +1,10 @@
 terraform {
-  required_version = ">= 1.0"
+  required_version = ">= 1.0.0"
```
Comment (korenyoni) [inline versions.tf]: I don't think any of our other modules support `1.0.0` as the bare minimum. Only components have this version requirement.

https://github.com/search?q=org%3Acloudposse+%22%3E%3D+1.0.0%22+filename%3Aversions.tf&type=Code&ref=advsearch&l=&l=

**Review by korenyoni — COMMENTED** [review]
See suggestion and also note on version requirement

**Review by korenyoni — DISMISSED** [review]

File: LICENSE [hunk 4]
```diff
@@ -178,15 +178,15 @@
...
+      boilerplate notice, with the fields enclosed by brackets "{}"
       replaced with your own identifying information. (Don't include
       the brackets!)  The text should be enclosed in the appropriate
       comment syntax for the file format. We also recommend that a
       file or class name and description of purpose be included on the
       same "printed page" as the copyright notice for easier
       identification within third-party archives.
 
-   Copyright 2020 Cloud Posse, LLC
+   Copyright 2017-2020 Cloud Posse, LLC
```
Comment (aknysh) [inline LICENSE]:
```suggestion
   Copyright 2020-2022 Cloud Posse, LLC
```

File: main.tf [hunk 5]
```diff
@@ -1,13 +1,50 @@
-resource "random_integer" "example" {
+locals {
+  principals_full_access_non_empty = length(var.principals_full_access) > 0 ? true : false
+  ecr_need_policy                  = length(var.principals_full_access) + length(var.principals_readonly_access) > 0 ? true : false
+}
+
+resource "aws_ecrpublic_repository" "this" {
+  for_each        = { for o in module.this.enabled ? var.repository_configs : [] : o.name => o }
```
Comment (aknysh) [inline main.tf]:
```suggestion
  for_each        = { for o var.repository_configs : o.name => o if module.this.enabled }
```

**Review by aknysh — COMMENTED** [review]
please see comments

**Review by aknysh — APPROVED** [review]

**Pull request merged**